cluster/saltbase/salt/kubelet/default

```diff
@@ -138,9 +138,9 @@
 {% if pillar.get('network_provider', '').lower() == 'opencontrail' %}
   {% set network_plugin = "--network-plugin=opencontrail" %}
 {% elif pillar.get('network_provider', '').lower() == 'cni' %}
-  {% set network_plugin = "--network-plugin=cni --network-plugin-dir=/etc/cni/net.d/" %}
+  {% set network_plugin = "--network-plugin=cni --network-plugin-dir=/etc/cni/net.d/ --cni-bin-dir=/opt/cni/bin" %}
 {%elif pillar.get('network_policy_provider', '').lower() == 'calico' and grains['roles'][0] != 'kubernetes-master' -%}
-  {% set network_plugin = "--network-plugin=cni --network-plugin-dir=/etc/cni/net.d/" %}
+  {% set network_plugin = "--network-plugin=cni --network-plugin-dir=/etc/cni/net.d/ --cni-bin-dir=/opt/cni/bin" %}
```

Comment:
> What we could do is pass --cni-bin-dir=/home/kubernetes/cni for all OSes 

That would be my preferred outcome. 

My only reservation is potentially breaking other CNI plugin DaemonSets.  

CC @kubernetes/sig-network-misc